strongbox-metadata-maven-api/src/main/java/org/carlspring/strongbox/storage/metadata/MetadataMerger.java

```diff
@@ -108,19 +128,96 @@ public Metadata updateMetadataAtGroupLevel(PluginArtifact artifact, Metadata met
         }
         if (!found)
         {
-            Plugin plugin = new Plugin();
-            // TODO: Figure out how to get plugin name from artifact
-            plugin.setName("");
-            plugin.setArtifactId(artifact.getArtifactId());
-            // TODO: Figure out how to get plugin prefix from artifact
-            plugin.setPrefix("");
-
+            Plugin plugin = null;
+            plugin = getPluginFromArtifact(artifact);
             plugins.add(plugin);
         }
         return metadata;
     }
 
-    private Collection<SnapshotVersion> createNewSnapshotVersions(String version, String timestamp, int buildNumber)
+    /**
+     * This method will read the artifact and find the plugin.xml in the artifact archive
+     * It then reads plugin.xml using SAX parser and extracts information.
+     *
+     * @param artifact
+     * @return
+     */
+    private Plugin getPluginFromArtifact(PluginArtifact artifact)
+    {
+        Plugin plugin = new Plugin();
+        HashMap<String, String> pluginMap = extractAndReadPluginXmlFromArtifact(artifact);
+
+        plugin.setName(pluginMap.get("name"));
+        plugin.setPrefix(pluginMap.get("goalPrefix"));
+        plugin.setArtifactId(artifact.getArtifactId());
+        return plugin;
+    }
+
+    /**
+     * This method iterates over the archived artifact until it finds the plugin.xml
+     * and calls the method extractPluginXmlFile() if the plugin.xml
+     * is found for extraction
+     *
+     * @param artifact
+     */
+    private HashMap<String, String> extractAndReadPluginXmlFromArtifact(PluginArtifact artifact)
+    {
+
+        FileInputStream fileInputStream = null;
+        ZipInputStream zipInputStream;
+        ZipEntry zipEntry;
+        HashMap<String, String> pluginMap = null;
+        try
+        {
+            fileInputStream = new FileInputStream(artifact.getFile().getCanonicalPath());
+            zipInputStream = new ZipInputStream(fileInputStream);
+
+
+            while ((zipEntry = zipInputStream.getNextEntry()) != null)
+            {
+                if (zipEntry.getName().endsWith("plugin.xml"))
```

Comment:
I think there should be a more explicit check for the exact path to the `META-INF/maven/plungin.xml`, because... what will happen, if, say, there is coincidentally some other `plugin.xml` file...?

Reply:
Replaced the check for plugin.xml with a more comprehensive check for plugin.xml path under META-INF/maven/plugin.xml